driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java

```diff
@@ -956,7 +975,11 @@ private boolean isValidPeer(Row peerRow, boolean logIfInvalid) {
               && !peerRow.isNull("native_port");
     } else {
       isValid &=
-          peerRow.getColumnDefinitions().contains("rpc_address") && !peerRow.isNull("rpc_address");
+          (peerRow.getColumnDefinitions().contains("rpc_address") && !peerRow.isNull("rpc_address"))
+              || (peerRow.getColumnDefinitions().contains("native_transport_address")
+                  && peerRow.getColumnDefinitions().contains("native_transport_port")
+                  && !peerRow.isNull("native_transport_address")
+                  && !peerRow.isNull("native_transport_port"));
```

Comment:
How about a row that has `native_transport_port_ssl` but not `native_transport_port`? Can this even happen?

Reply:
native_transport_port has a default value of 9042 so even if nothing is specified for it in cassandra.yaml the corresponding system.peers row will still have a value.  I don't think there is a way to create a row here that doesn't have a native_transport_port entry so this logic should hold up.  I'll take a look at DSE code in a bit to confirm.

I also just fired up my local 3 node DSE 6.8.13 cluster to confirm... and it does look like the value is populated as expected.